From 48e9cf69716bbee48ed1e93e551f377eec01957b Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Sun, 2 Feb 2020 13:35:20 +0200
Subject: [PATCH 22/22] Makefile.inc1: Pass _SHLIBDIRPREFIX for btools too.

 Allow to to link in with crossworld target built libraries if available.

 These are not needed for DragonFly crossworld target, however helps with
 bootstrapping on OpenBSD or glibc based systems where static versions
 are not available or mismatches the lex(1) too much.
---
 Makefile.inc1            | 1 +
 usr.bin/gzip/Makefile    | 4 ++++
 usr.bin/m4/Makefile      | 5 ++++-
 usr.sbin/config/Makefile | 4 ++++
 4 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/Makefile.inc1 b/Makefile.inc1
index 4e2ac9d01a..514543c873 100644
--- a/Makefile.inc1
+++ b/Makefile.inc1
@@ -227,6 +227,7 @@ BMAKEENV=	MAKEOBJDIRPREFIX=${BTOOLSDEST} \
 		OBJTREE=${OBJTREE} \
 		DESTDIR=${BTOOLSDEST} \
 		LC_ALL=C \
+		_SHLIBDIRPREFIX=${BTOOLSDEST} \
 		PATH=${BTOOLSPATH}:${PATH} \
 		M4=${BTOOLSDEST}/usr/bin/m4 \
 		INSTALL="sh ${.CURDIR}/tools/install.sh"
diff --git a/usr.bin/gzip/Makefile b/usr.bin/gzip/Makefile
index 9fe453a004..a40712a8b4 100644
--- a/usr.bin/gzip/Makefile
+++ b/usr.bin/gzip/Makefile
@@ -4,6 +4,10 @@
 PROG=		gzip
 MAN=		gzip.1 gzexe.1 zdiff.1 zforce.1 zgrep.1 zmore.1 znew.1
 
+.if defined(BOOTSTRAPPING) && exists(${_SHLIBDIRPREFIX}/usr/lib/libz.a)
+LDFLAGS+=	-L${_SHLIBDIRPREFIX}/usr/lib
+.endif
+
 DPADD=		${LIBZ}
 LDADD=		-lz
 
diff --git a/usr.bin/m4/Makefile b/usr.bin/m4/Makefile
index 8cc7535ffb..a68f096f26 100644
--- a/usr.bin/m4/Makefile
+++ b/usr.bin/m4/Makefile
@@ -1,11 +1,14 @@
 #	$OpenBSD: Makefile,v 1.10 2002/04/26 13:13:41 espie Exp $
 # $FreeBSD: src/usr.bin/m4/Makefile,v 1.14 2013/01/01 18:26:42 svnexp Exp $
 
-# -DEXTENDED 
+# -DEXTENDED
 # 	if you want the paste & spaste macros.
 
 PROG=	m4
 CFLAGS+=-DEXTENDED -I${.CURDIR} -I${.CURDIR}/lib -I.
+.if defined(BOOTSTRAPPING) && exists(${_SHLIBDIRPREFIX}/usr/lib/libl.a)
+LDFLAGS+=	-L${_SHLIBDIRPREFIX}/usr/lib
+.endif
 DPADD=	${LIBY} ${LIBL} ${LIBM}
 LDADD=	-ly -ll -lm
 
diff --git a/usr.sbin/config/Makefile b/usr.sbin/config/Makefile
index 61b55cef5a..6133a478c9 100644
--- a/usr.sbin/config/Makefile
+++ b/usr.sbin/config/Makefile
@@ -9,6 +9,10 @@ MAN=	config.8
 DPADD=	${LIBL}
 LDADD=	-ll
 
+.if defined(BOOTSTRAPPING) && exists(${_SHLIBDIRPREFIX}/usr/lib/libl.a)
+LDFLAGS+=	-L${_SHLIBDIRPREFIX}/usr/lib
+.endif
+
 # XXX: for now do not let the use of -flto, gcc80
 .if ${CFLAGS:M-flto}
 CFLAGS+=	-fno-lto
-- 
2.25.0

